src/main/java/io/stargate/sgv2/jsonapi/service/updater/DocumentUpdater.java

```diff
@@ -105,9 +132,65 @@ private boolean replace(ObjectNode docToUpdate, boolean docInserted) {
     return true;
   }
 
+  /**
+   * vectorize replacementDocument as needed, only when document is found will be used by
+   * findOneAndReplace
+   *
+   * @param dataVectorizer
+   */
+  public Uni<Boolean> vectorizeTheReplacementDocument(DataVectorizer dataVectorizer) {
```

Comment:
nit: we don't usually use articles in method names, so `vectorizeReplacementDocument()`.

Reply:
Thanks! fixed